File: go/valid-palindrome/valid-palindrome.go
package main

import(
	"fmt"
	"strings"
	"unicode"
)

func cleanString(input string) string {
	var builder strings.Builder

	for _, char := range input {
		if unicode.IsLetter(char) || unicode.IsDigit(char) {
			builder.WriteRune(char)
		}
	}

	return strings.ToLower(builder.String())
}
func isPalindrome(s string) bool{
	sClean := cleanString(s)
	l := 0
	r := len(sClean)-1

	for l < r{
		if sClean[l] != sClean[r]{
			return false
		} 
		l++
		r--
	}
	return true
}


func main(){
	s := "A man, a plan, a canal: Panama"
	fmt.Printf("the string %s is palindrome: %v \n",s ,isPalindrome(s))
}

File: js/valid-palindrome/valid-palindrome.js
function cleanString(input) {
    return input.replace(/[^a-zA-Z0-9]/g, '');
}

function validPalindrome(s){
    let sClean = cleanString(s).toLowerCase().split("")
    let l = 0 , r = sClean.length
    for (let i =0; i < Math.floor(sClean.length/2); i++){
        if(sClean[l] != sClean[r-1]){
            return false
        }
        l++
        r--
        
    }
    return true
}   

let s = "A man, a plan, a canal: Panama"

console.log("answer to the first example is:",validPalindrome(s))